ml-agents/mlagents/trainers/saver/tf_saver.py

```diff
@@ -0,0 +1,134 @@
+import os
+from mlagents_envs.exception import UnityPolicyException
+from mlagents_envs.logging_util import get_logger
+from mlagents.tf_utils import tf
+from mlagents.trainers.saver.saver import BaseSaver
+from mlagents.trainers.tf.model_serialization import export_policy_model
+from mlagents.trainers.settings import TrainerSettings
+from mlagents.trainers.policy.tf_policy import TFPolicy
+from mlagents.trainers import __version__
+
+
+logger = get_logger(__name__)
+
+
+class TFSaver(BaseSaver):
+    """
+    Saver class for TensorFlow
+    """
+
+    def __init__(
+        self,
+        policy: TFPolicy,
+        trainer_settings: TrainerSettings,
+        model_path: str,
+        load: bool = False,
+    ):
+        super().__init__()
+        self.policy = policy
+        self.model_path = model_path
+        self.initialize_path = trainer_settings.init_path
+        self._keep_checkpoints = trainer_settings.keep_checkpoints
+        self.load = load
+
+        self.graph = self.policy.graph
+        self.sess = self.policy.sess
+        with self.graph.as_default():
+            self.saver = tf.train.Saver(max_to_keep=self._keep_checkpoints)
+
+    def register(self, module):
```

Comment:
What do you think about making the `self.policy` and `self.graph`, etc. assigned in the `register` method? We could do `if isinstance(module, TFPolicy)` and then set those equal if so. Otherwise, pass. 

This would unify the constructor and `register` methods between TFSaver and TorchSaver, where it looks like currently the `self.policy` is not used by TorchSaver and `register` is not used by TFSaver.

Reply:
Once I was thinking doing in that way. The pros are as you said. The cons: having more dummy `register` functions in TF optimizer class; and `if isinstance()` needed in `register` for both TFSaver and TorchSaver. TorchSaver also needs the policy for the purpose of setting global step.
I'm fine with both approach actually, went with current one just because it's simpler.

Reply:
Actually after more thinking I feel that the way you said is better. Will change this.